relay-server/src/envelope.rs

```diff
@@ -136,32 +139,88 @@ impl ContentType {
             Self::OctetStream => "application/octet-stream",
             Self::Minidump => "application/x-dmp",
             Self::Xml => "text/xml",
+            Self::Envelope => self::CONTENT_TYPE,
             Self::Other(ref other) => &other,
         }
     }
 
-    fn from_str(content_type: &str) -> Option<Self> {
-        match content_type {
-            "text/plain" => Some(Self::Text),
-            "application/json" => Some(Self::Json),
-            "application/x-msgpack" => Some(Self::MsgPack),
-            "application/x-dmp" => Some(Self::Minidump),
-            "text/xml" | "application/xml" => Some(Self::Xml),
-            "application/octet-stream" => Some(Self::OctetStream),
-            _ => None,
+    fn from_str(ct: &str) -> Option<Self> {
+        if ct.eq_ignore_ascii_case(Self::Text.as_str()) {
+            Some(Self::Text)
+        } else if ct.eq_ignore_ascii_case(Self::Json.as_str()) {
+            Some(Self::Json)
+        } else if ct.eq_ignore_ascii_case(Self::MsgPack.as_str()) {
+            Some(Self::MsgPack)
+        } else if ct.eq_ignore_ascii_case(Self::OctetStream.as_str()) {
+            Some(Self::OctetStream)
+        } else if ct.eq_ignore_ascii_case(Self::Minidump.as_str()) {
+            Some(Self::Minidump)
+        } else if ct.eq_ignore_ascii_case(Self::Xml.as_str())
+            || ct.eq_ignore_ascii_case("application/xml")
+        {
+            Some(Self::Xml)
+        } else if ct.eq_ignore_ascii_case(Self::Envelope.as_str()) {
+            Some(Self::Envelope)
+        } else {
+            None
```

Comment:
This looks even more gross than before. Could we possibly match on `content_type.to_ascii_lowercase()`?

Reply:
No, that is sadly not possible. The only way would be to match on `.to_lowercase()` which would allocate.

Reply:
Coding principle. One day, I'll document that in our coding guide :)